docker/README.md

```diff
@@ -246,6 +256,29 @@ with the PDO tool `pdo-configure-services`.
         --name ${USER}_services_container pdo_services --mode copy
 ```
 
+#### PDO Services Deployment Using SGX ####
+
+There are a few _additional_ considerations when using the services with SGX.
+
+Before starting the container, make sure that the SGX collateral is available 
+as described [here](../docs/install).
+
+Also, recall that the attestation policy on the ledger has to be set once by the 
+first eservice of a ledger consortium member. Hence, the first service container 
+that is deputed to perform such registration must be instructed to do so.
+```bash
+    docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host \
+        --name ${USER}_services_container pdo_services --register
+```
+This updated command allows to trigger the registration step right before
+starting the services. The policy registration must happen before enclaves are 
+registered (or any enclave registration will fail).
+
+Finally, the _same_ SGX collateral must be made available to all service containers.
+At enclave registration time, this will allow the eservice to generate the right 
+quote (and attestation verification report) that meets the attestation policy 
+originally registered with the PDO Transaction Processor.
```

Comment:
We should say here also where this information is found and also should mention (_and use_!) the defaults of `build/keys/sgx_mode_{sim|hw)`.  Besides not supporting any default and requiring `PDO_SGX_KEY_ROOT` to be defined when running make docker, having to define it explicitly is also conceptually inconsistent as this variable is (at least as far as our existing default directories goes) dependent on `SGX_MODE`.  So we are using the worst of the combination: mode is now managed by targets yet we still need mode-dependent external env variables, not allowing for a no-config version of `make -C docker` (as previously was possible; admittedly previously only for SGX_MODE=sim but now as we have different target we could retain that also for SGX_MODE=hw)

Reply:
> We should say here also where this information is found and also should mention (and use!) the defaults of build/keys/sgx_mode_{sim|hw).

Correct, and that is described under `docs/install.md`.

> Besides not supporting any default and requiring PDO_SGX_KEY_ROOT to be defined when running make docker, having to define it explicitly is also conceptually inconsistent as this variable is (at least as far as our existing default directories goes) dependent on SGX_MODE.

Let me refer again to the documentation in `docs/host_install.md`, which indicates to source `build/common-config.sh` first. If `PDO_SGX_KEY_ROOT` is not defined, this steps creates a default value for it as you mentioned.

> So we are using the worst of the combination: mode is now managed by targets yet we still need mode-dependent external env variables

This is what was discussed -- namely to have specific SGX targets (which clearly manage the mode).
Again, the "external" variables are not necessary if `build/common-config.sh` is sourced first -- as it creates the default values.

I agree that the combination can be a problem when switching on the fly between SIM and HW mode development.
For this reason the BKM is to always start with fresh clean environment.
I'm open to suggestions to simplify this for developers.

Reply:
> . I'm open to suggestions to simplify this for developers.

Well, isn't typing in the image names exactly a simple way to simplify it & make it more robust?

Reply:
It probably is -- but image typing brings us back to the other [thread](https://github.com/hyperledger-labs/private-data-objects/pull/477#discussion_r1538575499).

Reply:
I have added a commit to type the service image built for sgx.